### PR TITLE
SONAR-5684 Sort rules by repository and key in backup file

### PR DESCRIPTION
**Pull request by jblievremont**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileBackuper.java [hunk 1]
```diff
@@ -191,4 +195,16 @@ private SMInputFactory initStax() {
     xmlFactory.setProperty(XMLInputFactory.IS_VALIDATING, Boolean.FALSE);
     return new SMInputFactory(xmlFactory);
   }
+
+  private enum BackupActiveRuleComparator implements Comparator<ActiveRule> {
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileBackuper.java]: What's the purpose of creating an enum ?

**Comment by julienlancelot** [comment]
Please answer my question

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileBackuper.java (hunk 1 above)
Comment (jblievremont) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileBackuper.java]: Saw this in @sns-seb 's code and liked it immediately: the most concise way to declare a static singleton class with implementation in Java - see e.g http://stackoverflow.com/questions/26285520/implementing-singleton-as-enum-in-java

Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileBackuper.java]: Ok, thanks for the pointer, it's indeed interesting.

**Comment by julienlancelot** [comment]
LGTM

**Pull request closed without merging**